scrapypuppeteer/middleware.py

```diff
@@ -149,12 +149,13 @@ def process_response(self, request, response, spider):
         if puppeteer_request is None:
             return response
 
-        if b'application/json' not in response.headers.get(b'Content-Type', b''):
-            return response.replace(request=request)
-
```

Comment:
We still have some non-json responses, e.g. 400 for missing parameters. Move them to json or leave their handling intact.